nomadproxy.go

```diff
@@ -132,12 +147,59 @@ func main() {
 	}
 
 	log.Printf("nomadproxy running at %v, proxying to %v", ln.Addr(), *backendAddr)
-	log.Fatal(http.Serve(ln, auditRequests(localClient, proxy)))
+	log.Fatal(http.Serve(ln, auditRequests(ctx, localClient, proxy)))
+}
+
+func newOauthClient() *tailscale.Client {
+	tailnet := os.Getenv("TAILNET_NAME")
+	clientID := os.Getenv("TS_OAUTH_CLIENT_ID")
+	clientSecret := os.Getenv("TS_OAUTH_CLIENT_SECRET")
+
+	if tailnet == "" || clientID == "" || clientSecret == "" {
+		log.Fatal("Missing configuration (TAILNET_NAME, TS_OAUTH_CLIENT_ID, TS_OAUTH_CLIENT_SECRET)")
+	}
+
+	return &tailscale.Client{
+		Tailnet: tailnet,
+		HTTP: tailscale.OAuthConfig{
+			ClientID:     clientID,
+			ClientSecret: clientSecret,
+			Scopes:       []string{"auth_keys:write"},
+		}.HTTPClient(),
+	}
+}
+
+func createAuthKey(ctx context.Context, client *tailscale.Client, tags []string) (string, error) {
+	req := tailscale.CreateKeyRequest{
+		Capabilities: tailscale.KeyCapabilities{
+			Devices: struct {
+				Create struct {
+					Reusable      bool     `json:"reusable"`
+					Ephemeral     bool     `json:"ephemeral"`
+					Tags          []string `json:"tags"`
+					Preauthorized bool     `json:"preauthorized"`
+				} `json:"create"`
+			}{},
+		},
+		ExpirySeconds: 60,
+	}
+
+	req.Capabilities.Devices.Create.Ephemeral = true
+	req.Capabilities.Devices.Create.Reusable = false
+	req.Capabilities.Devices.Create.Preauthorized = false
+	req.Capabilities.Devices.Create.Tags = tags
```

Comment:
I'm not sure why you're re-defining the contents of the structs here?

```suggestion
req := tailscale.CreateKeyRequest{
	Capabilities: tailscale.KeyCapabilities{
		Devices: tailscale.KeyDeviceCapabilities{
			Create: tailscale.KeyDeviceCreateCapabilities{
				Ephemeral: true,
				Reusable: true,
				Preauthorized: false,
				Tags: tags,
			},
		},
		ExpirySeconds: 60,
	}
}
```

should be enough here

Reply:
Ah, it looks like the tailscale v2 pkg no longer defines structs for `Devices` or `Create`.  Here's the [KeyCapabilities struct](https://github.com/tailscale/tailscale-client-go/blob/74c8fc3cb4d7/v2/keys.go#L18-L27).  (I do see them in the deprecated v1 version.... wonder why they chose to forgo?)

Is there another/cleaner way to go about it when using the v2 client pkg?

Reply:
https://github.com/tailscale/tailscale-client-go/blob/main/tailscale/client_test.go#L767-L771 would work

Reply:
Thank you.  Updated.